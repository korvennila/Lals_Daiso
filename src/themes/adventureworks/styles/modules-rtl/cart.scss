@import "../common/00-settings/responsive-breakpoints.scss";

$cart-content-margin-right: 0;
$cart-content-mobile-vp-margin-right: 0;
$cart-remove-item-margin-right: 20px;
$cart-content-margin-left-zero: 0;
$cart-promotion-text-padding-right: 24px;
$rtl-cart-bopis-heading-icon-padding-right: 30px;
$rtl-cart-bopis-heading-margin-right: 76px;
$rtl-cart-lines-heading-title-margin-right: 76px;
$rtl-cart-line__product-quantity-decrement-margin-left: 7px;
$rtl-cart-line__product-quantity-decrement-margin-right: 0;
$rtl-cart-line__product-quantity-increment-margin-left: 0;
$rtl-cart-line__product-quantity-increment-margin-right: 7px;
$rtl-cart-line__product-savings-margin-right: 5px;
$rtl-cart-lines-heading-icon-padding-right-mobile: 18px;
$rtl-cart-lines-heading-margin-right-mobile: 62px;
$rtl-cart-lines-heading-title-margin-right-mobile: 62px;

[dir="rtl"] {
    .ms-cart {
        .msc-cart-lines-group {
            margin-left: 21px;
            margin-right: 0;

            .msc-cart-lines-group-wraper {
                .msc-cart-lines-group-wrapper__bopis-heading {
                    &-pickup-icon,
                    &-shipping-icon,
                    &-pay-invoices-icon {
                        padding-left: 0;
                        padding-right: $rtl-cart-bopis-heading-icon-padding-right;
                    }

                    &-pickup,
                    &-shipping,
                    &-pay-invoices {
                        margin-left: 0;
                        margin-right: $rtl-cart-bopis-heading-margin-right;
                    }

                    &-title {
                        margin-left: 0;
                        margin-right: $rtl-cart-lines-heading-title-margin-right;
                    }
                }

                .msc-cart-line__product-quantity {
                    .decrement {
                        margin-left: $rtl-cart-line__product-quantity-decrement-margin-left;
                        margin-right: $rtl-cart-line__product-quantity-decrement-margin-right;
                    }

                    .increment {
                        margin-left: $rtl-cart-line__product-quantity-increment-margin-left;
                        margin-right: $rtl-cart-line__product-quantity-increment-margin-right;
                    }
                }

                .msc-cart-line__product-savings {
                    .msc-cart-line__product-savings-text {
                        margin-left: 0;
                        margin-right: $rtl-cart-line__product-savings-margin-right;
                    }
                }
            }
        }

        .msc-cart-line {
            &-group__extra-actions,
            &__content,
            &__extra-actions,
            &__actions {
                margin-right: $cart-content-margin-right;
                margin-left: $cart-content-margin-left-zero;
            }

            &__invoice-image {
                margin-right: 0;
                margin-left: 20px;
            }

            .msc-cart-line__product-image {
                .msc_image {
                    transform: matrix(-1, 0, 0, 1, 0, 0);
                }
            }
        }

        .msc-cart__empty-cart {
            .msc-cart-line {
                .ms-content-block {
                    .ms-content-block__details {
                        .ms-content-block__title {
                            text-align: center;
                        }

                        .ms-content-block__text {
                            text-align: center;
                        }
                    }
                }
            }
        }

        .msc-cart-lines-item > .msc-cart-line__add-to-wishlist ~ .msc-cart-line__remove-item {
            margin-right: auto;
            margin-left: unset;
        }

        /* stylelint-disable-next-line no-descending-specificity -- Auto-suppressed. Please fix if possible. */
        .msc-cart-lines-item > .msc-cart-line__add-to-wishlist, .msc-cart-lines-item > .msc-cart-line__remove-item {
            margin-right: $cart-content-margin-right;
            margin-left: unset;
        }

        .msc-cart__btn-addcarttotemplate-tooltip {
            .msc-tooltip-inner {
                text-align: right;
            }
        }

        .msc-order-summary-wrapper {
            .msc-order-summary__heading {
                padding: 31px 31px 20px 0;
            }

            .msc-order-summary__checkout .msc-alert-danger {
                @media (min-width: $msv-breakpoint-l) {
                    margin-right: 32px;
                }
            }
        }

        .msc-order-summary__checkout .msc-promo-code-heading {
            padding: 30px 31px 25px 0;
        }

        .msc-cart__promotion {
            span {
                padding-right: $cart-promotion-text-padding-right;
                padding-left: 0;
                float: right;
            }
        }
    }

    @media screen and (max-width: $msv-breakpoint-m - 1) {
        .ms-cart {
            .msc-cart-lines-group {
                margin-left: 0;
                margin-right: 0;
            }

            .msc-cart-lines-group-wraper {
                .msc-cart-lines-group-wrapper__bopis-heading {
                    &-pickup-icon,
                    &-shipping-icon,
                    &-pay-invoices-icon {
                        padding-left: 0;
                        padding-right: $rtl-cart-lines-heading-icon-padding-right-mobile;
                    }

                    &-pickup,
                    &-shipping,
                    &-pay-invoices {
                        margin-left: 0;
                        margin-right: $rtl-cart-lines-heading-margin-right-mobile;
                    }

                    &-title {
                        margin-left: 0;
                        margin-right: $rtl-cart-lines-heading-title-margin-right-mobile;
                    }
                }
            }

            .msc-order-summary__checkout .msc-promo-code-heading {
                padding-right: 16px;
                padding-left: 12px;
            }

            .msc-order-summary-wrapper {
                .msc-order-summary__heading {
                    padding-right: 16px;
                    padding-left: 12px;
                }
            }

            .msc-cartline-wraper .msc-cart-lines-button-container .msc-cart-line__actions {
                padding: 0;
            }

            .msc-cart-line__product {
                margin-right: 100px;
                margin-left: 0;
            }

            .msc-cart-line {
                &__content,
                &__bopis-container,
                &__extra-actions {
                    margin-right: $cart-content-mobile-vp-margin-right;
                    margin-left: $cart-content-margin-left-zero;
                }

                &__product-savings {
                    margin-left: unset;
                    margin-right: 0;
                    text-align: right;
                }
            }

            .msc-cart-lines-item > .msc-cart-line__add-to-wishlist,
            .msc-cart-lines-item > .msc-cart-line__remove-item {
                margin-left: unset;
                margin-right: $cart-content-mobile-vp-margin-right;
            }

            .msc-order-summary__checkout-button {
                .msc-cart__btn-checkout {
                    margin-right: 0;
                }
            }

            .msc-cart__promotion {
                span {
                    margin-right: 0;
                    margin-left: 30px;
                }
            }
        }
    }

    @media (min-width: $msv-breakpoint-m) and (max-width: $msv-breakpoint-l) {
        .ms-cart {
            .msc-cart-lines-group {
                margin-left: 0;
                margin-right: 0;

                &-wraper {
                    .msc-cart-lines-item {
                        .msc-cart-line__content {
                            .msc-cart-line__product {
                                margin-right: 0;
                                margin-left: 10px;
                            }
                        }
                    }
                }
            }

            .msc-order-summary__checkout .msc-promo-code-heading {
                padding-right: 16px;
                padding-left: 12px;
            }

            .msc-order-summary-wrapper .msc-order-summary__heading {
                padding-right: 16px;
                padding-left: 12px;
            }

            .msc-order-summary__checkout-button {
                .msc-cart__btn-checkout {
                    margin-right: 0;
                    margin-left: 12px;
                }

                .msc-cart__btn-guestcheckout {
                    margin-left: 0;
                    margin-right: 12px;
                }
            }

            .msc-order-summary-wrapper {
                .msc-order-summary__checkout .msc-cart__btn-guestcheckout.msc-btn {
                    margin-left: 0;
                    margin-right: 12px;
                }
            }
        }
    }
}
