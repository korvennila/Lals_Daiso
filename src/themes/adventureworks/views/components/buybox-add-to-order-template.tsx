/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import {
    AddToOrderTemplateComponent,
    IAddToOrderTemplateDialogResources,
    IDuplicateItemsWhenAddingToOrderTemplateDialogResources,
    IItemAddedToOrderTemplateDialogResources,
    IOrderTemplateActionErrorResult,
    IOrderTemplateNameDialogResources
} from '@msdyn365-commerce/components';
import { SimpleProduct } from '@msdyn365-commerce/retail-proxy';
import {
    getConfigureErrors,
    IBuyboxCallbacks,
    IBuyboxCommonResources,
    IBuyboxData,
    IBuyboxState,
    IBuyboxViewProps
} from '@msdyn365-commerce-modules/buybox';
import * as React from 'react';

import { IBuyboxProps as IBuyboxExtentionProps } from '../../definition-extensions/buybox.ext.props.autogenerated';

/**
 * To generate AddToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of AddToOrderTemplateDialog resources..
 */
function generateAddToOrderTemplateDialogResources(resources: IBuyboxCommonResources): IAddToOrderTemplateDialogResources {
    return {
        addToOrderTemplateHeader: resources.addToOrderTemplateHeader,

        noOrderTemplatesMessage: resources.noOrderTemplatesMessage,
        noOrderTemplatesDescription: resources.noOrderTemplatesDescription,

        createAnOrderTemplateButtonText: resources.createAnOrderTemplateButtonText,
        createNewOrderTemplateButtonText: resources.createNewOrderTemplateButtonText,
        cancelOrderTemplateCreationButtonText: resources.cancelOrderTemplateCreationButtonText,

        selectTemplatesText: resources.selectTemplatesText,
        addToTemplateButtonText: resources.addToTemplateButtonText,
        lineItemsText: resources.lineItemsText
    };
}

/**
 * To generate OrderTemplateNameDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of OrderTemplateNameDialog resources..
 */
function generateOrderTemplateNameDialogResources(resources: IBuyboxCommonResources): IOrderTemplateNameDialogResources {
    return {
        orderTemplateHeaderLabel: resources.createOrderTemplateHeader,
        orderTemplateTitleLabel: resources.orderTemplateTitle,
        orderTemplateNameAriaLabel: resources.orderTemplateNameAriaLabel,
        orderTemplateDescription: resources.createOrderTemplateDescription,

        defaultOrderTemplateName: resources.defaultOrderTemplateName,

        orderTemplateButtonLabel: resources.createOrderTemplateButtonText,
        orderTemplateCancelButtonLabel: resources.cancelNewOrderTemplateCreationButtonText
    };
}

/**
 * To generate ItemAddedToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of ItemAddedToOrderTemplateDialog resources..
 */
function generateItemAddedToOrderTemplateDialogResources(resources: IBuyboxCommonResources): IItemAddedToOrderTemplateDialogResources {
    return {
        viewOrderTemplateButtonText: resources.viewOrderTemplateButtonText,
        continueShoppingButtonText: resources.continueShoppingButtonText,
        itemAddedToOrderTemplateHeaderItemOneText: resources.itemAddedToOrderTemplateHeaderItemOneText,
        itemAddedToOrderTemplateHeaderItemFormatText: resources.itemAddedToOrderTemplateHeaderItemFormatText,
        itemAddedToOrderTemplateHeaderMessageText: resources.itemAddedToOrderTemplateHeaderMessageText,
        freePriceText: resources.priceFree,
        originalPriceText: resources.originalPriceText,
        currentPriceText: resources.currentPriceText
    };
}

/**
 * To generate DuplicateItemsWhenAddingToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of DuplicateItemsWhenAddingToOrderTemplateDialog resources..
 */
function generateDuplicateItemsWhenAddingToOrderTemplateDialogResources(
    resources: IBuyboxCommonResources
): IDuplicateItemsWhenAddingToOrderTemplateDialogResources {
    return {
        duplicatedProductsHeader: resources.duplicatedProductsHeader,
        duplicatedProductsDescription: resources.duplicatedProductsDescription,
        updateQuantityButtonText: resources.updateQuantityButtonText,
        cancelDuplicateItemsButtonText: resources.cancelDuplicateItemsButtonText
    };
}

/**
 * Add to wishlist failed function.
 * @param callbacks -Buybox callbacks.
 * @param resources -Buybox resources.
 * @param product -Simple product.
 * @returns Update error state.
 */
const onAddToOrderTemplateFailed = (callbacks: IBuyboxCallbacks, resources: IBuyboxCommonResources, product: SimpleProduct | undefined) => (
    result: IOrderTemplateActionErrorResult
) => {
    callbacks.updateErrorState({
        errorHost: 'ORDER_TEMPLATE',
        configureErrors:
            result.status === 'MISSING_DIMENSION' ? getConfigureErrors(result.missingDimensions, resources, product?.IsGiftCard) : {}
    });
};

/**
 * RenderAddToOrderTemplateButton.
 * @param props - Buybox view props.
 * @param state - Buybox state.
 * @param callbacks - Buybox callbacks.
 * @param product - Product data.
 * @returns -- Returns JSX element.
 */
export const renderAddToOrderTemplateButton = (
    props: IBuyboxViewProps & IBuyboxExtentionProps<IBuyboxData>,
    state: IBuyboxState,
    callbacks: IBuyboxCallbacks,
    product: SimpleProduct
): React.ReactNode => {
    const {
        // eslint-disable-next-line @typescript-eslint/naming-convention -- Dependency from module file.
        id,
        typeName,
        context,
        data: {
            orderTemplates: { result: orderTemplates },
            customerInformation: { result: customerInformation }
        },
        resources,
        config
    } = props;

    const { quantity } = state;

    const isShowButtonText = true;
    const isMultipleCatalogsFeatureEnabled = props.isChannelMultipleCatalogsFeatureEnabled || false;
    return (
        orderTemplates && (
            <AddToOrderTemplateComponent
                className='msc-add-to-cart-extra-actions'
                addToOrderTemplateButtonText={resources.addToOrderTemplateButtonText}
                addToOrderTemplateButtonTooltip={resources.addToOrderTemplateButtonTooltip}
                addToOrderTemplateDialogResources={generateAddToOrderTemplateDialogResources(resources)}
                createOrderTemplateDialogResources={generateOrderTemplateNameDialogResources(resources)}
                itemAddedToOrderTemplateDialogResources={generateItemAddedToOrderTemplateDialogResources(resources)}
                duplicateItemsWhenAddingToOrderTemplateDialogResources={generateDuplicateItemsWhenAddingToOrderTemplateDialogResources(
                    resources
                )}
                data={{ product, quantity, orderTemplates, customerInformation }}
                context={context}
                id={id}
                typeName={typeName}
                onError={onAddToOrderTemplateFailed(callbacks, resources, product)}
                shouldShowButtonFailedTooltip={state.isCustomPriceSelected}
                showButtonTooltip={false}
                showButtonText={isShowButtonText}
                imageSettings={config && config.imageSettings !== undefined ? config.imageSettings : undefined}
                isMultipleCatalogFeatureEnabledInHq={isMultipleCatalogsFeatureEnabled}
                {...(isMultipleCatalogsFeatureEnabled && { catalogId: props.currentCatalogId })}
            />
        )
    );
};
