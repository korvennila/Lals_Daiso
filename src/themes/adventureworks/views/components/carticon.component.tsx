/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { IComponent, IComponentProps, msdyn365Commerce } from '@msdyn365-commerce/core';
import { ICartState } from '@msdyn365-commerce/global-state';
import { format, getPayloadObject, getTelemetryAttributes, ITelemetryContent } from '@msdyn365-commerce-modules/utilities';
import { observer } from 'mobx-react';
import * as React from 'react';

/**
 * ICartIconData: Interface for Cart Icon Data.
 */
export interface ICartIconData {
    cart?: ICartState;
}

/**
 * ICartIconComponentProps: Interface for Cart Icon component props.
 */
export interface ICartIconComponentProps extends IComponentProps<ICartIconData> {
    className?: string;
    cartLabel: string;
    cartQtyLabel: string;
    telemetryContent?: ITelemetryContent;
    salesAgreementPricePrompt?: string;
}

/**
 * ICartIconComponent: Interface for Cart Icon component.
 */
export interface ICartIconComponent extends IComponent<ICartIconComponentProps> {}

const CartIconComponentActions = {};

/**
 *
 * CartIcon component.
 * @extends {React.FC<ICartIconComponentProps>}
 */
const CartIcon: React.FC<ICartIconComponentProps> = observer((props: ICartIconComponentProps) => {
    const {
        cartLabel,
        cartQtyLabel,
        data: { cart }
    } = props;
    const defaultCartItemCount: number = 0;

    const cartItem = cart ? cart.totalItemsInCart : defaultCartItemCount;
    let qtyLabel: string = '';
    const cartMaxQuantity: number = 99;
    const maxQuantityLabel: string = '99+';
    if (cartItem > cartMaxQuantity) {
        qtyLabel = maxQuantityLabel;
    } else {
        qtyLabel = format(cartQtyLabel, cartItem);
    }
    const label = format(cartLabel, cartItem);

    // Construct telemetry attribute to render
    const payLoad = getPayloadObject('click', props.telemetryContent!, 'cart-icon', '');
    const attributes = getTelemetryAttributes(props.telemetryContent!, payLoad);
    const style: React.CSSProperties = {
        visibility: cart ? 'visible' : 'hidden'
    };
    return (
        <div className='msc-cart-icon' role='button' aria-label={label} title={label} {...attributes}>
            <div className='msc-cart-icon__count' aria-hidden style={style}>
                {qtyLabel}
            </div>
        </div>
    );
});

export const CartIconComponent: React.FunctionComponent<ICartIconComponentProps> = msdyn365Commerce.createComponentOverride<
    // @ts-expect-error -- Compatible issue with the component override.
    ICartIconComponent
>('CartIcon', { component: CartIcon, ...CartIconComponentActions });

export default CartIconComponent;
