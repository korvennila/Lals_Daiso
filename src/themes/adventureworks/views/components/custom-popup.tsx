import React from 'react';

interface PopupProps {
    message: string;
    isVisible: boolean;
    onClose: () => void;
}

const CustomPopup: React.FC<PopupProps> = ({ message, isVisible, onClose }) => {
    if (!isVisible) return null;

    return (
        <div className='custom-popup-container'>
            <div className='custom-popup-content'>
                <p>{message}</p>
                <button onClick={onClose}>Close</button>
            </div>
            <div className='custom-popup-overlay' onClick={onClose}></div>
        </div>
    );
};

export default CustomPopup;
