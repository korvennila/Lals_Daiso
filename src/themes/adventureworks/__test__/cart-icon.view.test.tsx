/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { AddToCartBehavior } from '@msdyn365-commerce/components';
import { buildMockModuleProps, ICoreContext } from '@msdyn365-commerce/core';
import { ICartState } from '@msdyn365-commerce/global-state';
import { AsyncResult, SimpleProduct } from '@msdyn365-commerce/retail-proxy';
import { Cart, ICartData, ICartIconViewProps, ICartViewProps } from '@msdyn365-commerce-modules/cart';
import { mount, render } from 'enzyme';
import * as React from 'react';

import { ICartResources } from '../definition-extensions/cart.ext.props.autogenerated';
import { ICartIconProps as ICartIconExtensionProps, ICartIconResources } from '../definition-extensions/cart-icon.ext.props.autogenerated';
import CartIconView from '../views/cart-icon.view';

const mockContext: ICoreContext = {
    app: {
        config: {
            imageSettings: {
                viewports: {
                    xs: { q: 'w=80&h=94&m=6', w: 0, h: 0 },
                    sm: { q: 'w=148&h=174&m=6', w: 0, h: 0 },
                    lg: { q: 'w=148&h=174&m=6', w: 0, h: 0 }
                },
                lazyload: true
            },
            addToCartBehavior: AddToCartBehavior.showMiniCart
        },
        routes: {
            home: '/fedev',
            checkout: '/checkout'
        }
    },
    request: {
        // @ts-expect-error partial mock
        user: {
            isAuthenticated: true
        },

        // @ts-expect-error partial mock
        channel: {
            PickupDeliveryModeCode: '60'
        },
        app: {
            platform: {
                enableDefaultOrderQuantityLimits: 'none'
            }
        },
        gridSettings: { lg: { w: 1599 } }
    },

    // @ts-expect-error - Mocking context.
    cultureFormatter: {
        formatCurrency: jest.fn(price => `$${price}`),
        formatDate: jest.fn(date => `$${date.toLocaleString()}`)
    }
};

// @ts-expect-error partial mock
const mockResources: ICartResources & ICartIconResources = {
    totalPriceFormatString: 'totalPriceFormatString',
    continueShoppingButtonTitle: 'continueShoppingButtonTitle',
    subTotalMessage: 'subTotalMessage',
    item: 'item',
    items: 'items',
    emptyPrice: 'emptyPrice'
};

const mockCartData = {
    Quantity: 1,
    NetAmountWithoutTax: 25,
    NetPrice: 27,
    TaxAmount: 2.7,
    TotalAmount: 25,
    Id: '123',
    ChargeLines: [
        {
            ChargeCode: 'FREIGHT',
            Value: 20
        }
    ],
    CartLines: [
        {
            ItemId: 10,
            LineId: 10,
            ProductId: 10
        }
    ]
};

const mockCartState = {
    isProductAddedToCart: true,

    // @ts-expect-error
    cart: mockCartData as Cart,
    addProductToCart: jest.fn()
};

const mockEmptyCartState = {
    cart: {} as Cart,
    addProductToCart: jest.fn()
};

// @ts-expect-error partial mock
const mockData: ICartData = {
    // @ts-expect-error
    cart: {
        status: 'SUCCESS',
        result: mockCartState
    } as AsyncResult<ICartState>,
    products: ([] as unknown) as AsyncResult<SimpleProduct[]>
};

// @ts-expect-error partial mock
const mockEmptyData: ICartData = {
    // @ts-expect-error
    cart: {
        status: 'SUCCESS',
        result: mockEmptyCartState
    } as AsyncResult<ICartState>
};

// @ts-expect-error partial mock
const mockWithoutData: ICartData = {
    cart: {
        status: 'SUCCESS',
        result: undefined
    } as AsyncResult<ICartState>,
    products: ([] as unknown) as AsyncResult<SimpleProduct[]>
};

const mockSlots = {
    storeSelector: [],
    emptyCart: [],
    promoContentItem: [],
    paymentExpress: []
};

describe('Cart view unit tests - View', () => {
    it('renders correctly with 2 catline and cart mock data', () => {
        const moduleProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = buildMockModuleProps(
            {},
            {}
        ) as ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}>;
        moduleProps.resources = mockResources;
        const mockProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = {
            ...moduleProps,
            context: mockContext,
            className: 'className',
            cartIcon: 'Icon',
            data: mockData,
            slots: mockSlots,
            state: { isModalOpen: true },
            orderSummaryHeading: <div />,
            OrderSummaryWrapper: { className: 'node-class-OrderSummaryWrapper' },
            OrderSummaryItems: { className: 'msc-invoice-summary__items' },
            cartlines: [
                {
                    storeLocation: true,
                    cartline: <div />,
                    remove: <button />,
                    hasError: false,
                    data: {
                        // @ts-expect-error partial mock
                        product: {},
                        cartline: {}
                    }
                },
                {
                    cartline: <div />,
                    remove: <button />,
                    hasError: false,
                    data: {
                        // @ts-expect-error partial mock
                        product: {},
                        cartline: {}
                    }
                }
            ],
            FlyoutContainer: { className: 'node-class-FlyoutContainer' },
            CartlinesWrapper: { className: 'node-class-CartlinesWrapper' },
            totalPrice: '$0.00',
            checkoutAsSignInUserButton: <button />,
            checkoutAsGuestButton: <button />,
            goToCartButton: <button />,
            waitingComponent: 'Waiting',
            isCartLoading: false,
            removeItemClickHandler: jest.fn()
        };
        const component = render(<CartIconView {...mockProps} />);
        expect(component).toMatchSnapshot();

        const wrapper = mount(<CartIconView {...mockProps} />);
        const button = wrapper.find('button');
        expect(button).toBeDefined();
        button.at(0).simulate('click');
        wrapper.unmount();
    });

    it('renders correctly without catline and empty cart mock data', () => {
        const moduleProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = buildMockModuleProps(
            {},
            {},
            {},
            mockContext
        ) as ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}>;
        moduleProps.resources = mockResources;
        const mockProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = {
            ...moduleProps,
            className: 'className',
            cartIcon: 'Icon',
            data: mockEmptyData,
            slots: mockSlots,
            orderSummaryHeading: <div />,
            OrderSummaryWrapper: { className: 'node-class-OrderSummaryWrapper' },
            OrderSummaryItems: { className: 'msc-invoice-summary__items' },
            FlyoutContainer: { className: 'node-class-FlyoutContainer' },
            CartlinesWrapper: { className: 'node-class-CartlinesWrapper' },
            totalPrice: '$0.00',
            checkoutAsSignInUserButton: <button />,
            checkoutAsGuestButton: <button />,
            goToCartButton: <button />,
            waitingComponent: 'Waiting',
            isCartLoading: false,
            removeItemClickHandler: jest.fn()
        };
        const component = render(<CartIconView {...mockProps} />);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly with 1 catline and  without cart mock data', () => {
        const moduleProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = buildMockModuleProps(
            {},
            {}
        ) as ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}>;
        moduleProps.resources = mockResources;
        const mockProps: ICartViewProps & ICartIconViewProps & ICartIconExtensionProps<{}> = {
            ...moduleProps,
            className: 'className',
            cartIcon: 'Icon',
            data: mockWithoutData,
            slots: mockSlots,
            orderSummaryHeading: <div />,
            OrderSummaryWrapper: { className: 'node-class-OrderSummaryWrapper' },
            OrderSummaryItems: { className: 'msc-invoice-summary__items' },
            cartlines: [
                {
                    cartline: <div />,
                    remove: <button />,
                    hasError: false,
                    data: {
                        // @ts-expect-error partial mock
                        product: {},
                        cartline: {}
                    }
                }
            ],
            FlyoutContainer: { className: 'node-class-FlyoutContainer' },
            CartlinesWrapper: { className: 'node-class-CartlinesWrapper' },
            totalPrice: '$0.00',
            checkoutAsSignInUserButton: <button />,
            checkoutAsGuestButton: <button />,
            goToCartButton: <button />,
            waitingComponent: 'Waiting',
            isCartLoading: false,
            removeItemClickHandler: jest.fn()
        };
        const component = render(<CartIconView {...mockProps} />);
        expect(component).toMatchSnapshot();

        const result = mount(<CartIconView {...mockProps} />);
        const cartIconView = result.instance() as CartIconView;
        cartIconView.setState({
            isModalOpen: false
        });
        cartIconView.forceUpdate();
        cartIconView.componentWillUnmount();
        result.unmount();
    });
});
