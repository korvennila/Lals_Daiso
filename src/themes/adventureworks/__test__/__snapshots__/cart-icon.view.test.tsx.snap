// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart view unit tests - View renders correctly with 1 catline and  without cart mock data 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "",
      "data-m-t": "mock-type",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ms-cart-icon-wrapper msc-btn",
        },
        "children": Array [
          Node {
            "data": "Icon",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Cart view unit tests - View renders correctly with 2 catline and cart mock data 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "",
      "data-m-t": "mock-type",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ms-cart-icon-wrapper msc-btn",
        },
        "children": Array [
          Node {
            "data": "Icon",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Cart view unit tests - View renders correctly without catline and empty cart mock data 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "",
      "data-m-t": "mock-type",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ms-cart-icon-wrapper msc-btn",
        },
        "children": Array [
          Node {
            "data": "Icon",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
